src/aggregator/server/m3msg/options.go

```diff
@@ -70,12 +79,19 @@ func (o *options) InstrumentOptions() instrument.Options {
 	return o.instrumentOpts
 }
 
-func (o *options) SetM3msgConfiguration(value *coordinatorM3msg.Configuration) Options {
+func (o *options) SetM3MsgConfiguration(value *m3msg.Configuration) Options {
 	opts := *o
 	opts.m3msgConfiguration = value
 	return &opts
 }
 
-func (o *options) M3msgConfiguration() *coordinatorM3msg.Configuration {
+func (o *options) M3MsgConfiguration() *m3msg.Configuration {
 	return o.m3msgConfiguration
 }
+
+func (o *options) Validate() error {
```

Comment:
Should this also validate any required sampling configuration as well since that causes an error later on?